### PR TITLE
Added load/save features as per issue #20

### PR DESCRIPTION
**Pull request by Zeus3101**
It's my first issue so I'm not sure if it's the right way to go. Would appreciate all feedback!

Here's the summary:
- Added new file "utils.py" in common for misc functions like load/save and updated common/__init__.py
- Modified only DDPG.py for now to incorporate new load/save features (see commit description)
- Made some minor formatting changes

### CODE REVIEW

File: jigglypuffRL/common/utils.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import torch
+
+
+def save_params(algo, name=None, version=None):
+    if name is None:
+        name = algo.network_type
+
+    checkpoint = {'algo': name,
+                  'weights': algo.ac.state_dict()}
```
Comment (Sharad24) [inline jigglypuffRL/common/utils.py:9]: There are other params that need to be saved as well. Especially since its RL, other params are very important e.g. gamma, tau, actor and critic LRs. We don't have a lot of these yet in the class object but we plan to add them in the future. So can you change this to something like:
```
checkpoint = self.__dict__
checkpoint['algo'] = name
checkpoint['weights'] = algo.ac.state_dict()
```

File: jigglypuffRL/common/utils.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+def load_params(name, version=None):
+    try:
+        if version is None:
+            checkpoint = torch.load("{}.pt").format(name)
+        else:
+            checkpoint = torch.load("{}-{}.pt".format(name, version))
+    except FileNotFoundError:
+        raise Exception("Check name and version number again")
+    return checkpoint
```
Comment (Sharad24) [inline jigglypuffRL/common/utils.py:26]: Also, can you update the object params according to the new save function. I think you can do something like loop over the dictionary items and use `setattr`. 

The model weights will have to be loaded in a separate statement still though

File: jigglypuffRL/deeprl/ddpg/ddpg.py [hunk 3]
```diff
@@ -73,10 +81,15 @@ def __init__(
...
         self.update_interval = update_interval
+        self.save_interval = save_interval
         self.layers = layers
         self.tensorboard_log = tensorboard_log
         self.seed = seed
         self.render = render
+        self.network_type = network_type
+        self.pretrained = pretrained
+        self.save_name = save_name
+        self.save_version = save_version
```
Comment (Sharad24) [inline jigglypuffRL/deeprl/ddpg/ddpg.py:92]: `Nit`
Over here could you also add statements for assigning the load/save function to a class variable.
Something like:
`self.save = save_params`
`self.load = load_params`

File: jigglypuffRL/deeprl/ddpg/ddpg.py [hunk 4]
```diff
@@ -100,12 +113,18 @@ def __init__(
...
-    def create_model(self, network_type):
+    def create_model(self):
         self.ac = ActorCritic(
-            self.env, network_type, self.layers, noise_std=self.noise_std
-        ).to(self.device)
+            self.env, self.network_type, self.layers, noise_std=self.noise_std
+        )
+
+        if self.pretrained:
+            checkpoint = load_params(self.save_name, self.save_version)
```
Comment (Sharad24) [inline jigglypuffRL/deeprl/ddpg/ddpg.py:124]: `Nit`
Now over here we can call `self.load()`

File: jigglypuffRL/deeprl/ddpg/ddpg.py (hunk 3 above)
Comment (Sharad24) [reply jigglypuffRL/deeprl/ddpg/ddpg.py:92]: I guess this just adds the functionality of having the common save and load functions as object methods

File: jigglypuffRL/common/utils.py (hunk 2 above)
Comment (Zeus3101) [reply jigglypuffRL/common/utils.py]: Changed it. Model should be exactly reproduced (didn't use deepcopy for the model so I'm not sure if it's "exactly") with all parameters as I used algo.\_\_dict\_\_

File: jigglypuffRL/common/utils.py (hunk 1 above)
Comment (Zeus3101) [reply jigglypuffRL/common/utils.py]: Generalized it to save all inputted parameters. Will directly change the checkpoint dict.

**Comment by Zeus3101** [comment]
I've made the changes you've asked for. I'm not sure if saving all the parameters using the checkpoint dict is necessary or not. It shouldn't affect the training time much if we decrease the saving frequency.

**Comment by ajaysub110** [comment]
> I'm not sure if saving all the parameters using the checkpoint dict is necessary or not.

In RL, we often change parameters during training. Like learning rate, gamma, epsilon. So its better to save those so you can continue training from a point later on

File: jigglypuffRL/deeprl/ddpg/ddpg.py [hunk 5]
```diff
@@ -28,11 +28,18 @@ class DDPG:
...
-    :param render (boolean): if environment is to be rendered
-    :param device (str): device to use for tensor operations; 'cpu' for cpu and 'cuda' for gpu
+    :param seed: (int) seed for torch and gym
+    :param render: (boolean) if environment is to be rendered
+    :param device: (str) device to use for tensor operations; 'cpu' for cpu and 'cuda' for gpu
+    :param pretrained: (boolean) if model has already been trained
+    :param save_name: (str) model save name (if None, model hasn't been pretrained)
+    :param save_version: (int) model save version (if None, model hasn't been pretrained)
+    :param save: (function) save function to save model with all parameters
+    :param load: (function) load function to load pretrained model with all parameters
```
[the remark this answers is not in the document]
Comment (ajaysub110) [reply jigglypuffRL/deeprl/ddpg/ddpg.py:41]: The checkpoint param too I think

[the remark this answers is not in the document]
Comment (Zeus3101) [reply jigglypuffRL/deeprl/ddpg/ddpg.py:41]: Ok cool
